### PR TITLE
Make test-tidy check for alphabetical ordering of #![feature(...)] statements

### LINKED ISSUE #13954: Make test-tidy check for alphabetical ordering of #![feature(...)] statements
We try to keep our lists ordered:

``` rust
#![feature(box_patterns)]
#![feature(box_syntax)]
#![feature(conservative_impl_trait)]
#![feature(custom_derive)]
#![feature(nonzero)]
```

These only exist in files named `lib.rs`, and it would be nice to have `./mach test-tidy` verify that they are in alphabetical order.

### PR DESCRIPTION
**Pull request by ghost**
## Performs tidy-checks for alphabetical ordering of feature attributes in `lib.rs` files.

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [x] These changes fix #13954  (github issue number if applicable).

- [X] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13986)

### CODE REVIEW

**Comment by Wafflespeanut** [comment]
I'll take a look at the PR later. Please fix the files in the meantime.

**Comment by iamrohit7** [comment]
@Wafflespeanut Shall I just fix the tidy issues in my local branch and push them as changes?

**Comment by jdm** [comment]
Yep!

File: python/tidy/servo_tidy/tidy.py [hunk 1]
```diff
@@ -510,6 +513,28 @@ def check_rust(file_name, lines):
                       + decl_found.format(crate_name))
             prev_crate[indent] = crate_name
 
+        # check alphabetical order of feature attributes in lib.rs files
+        if is_lib_rs_file:
+            match = re.search(r"#!\[feature\((.*)\)\]", line)
+
+            if match:
+                features = map(lambda w: w.strip(), match.groups()[0].split(','))
```
Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: Nit: This could simply be `match.group(1).split(',')`

**Comment by Wafflespeanut** [comment]
LGTM aside from that tiny change :)

**Comment by iamrohit7** [comment]
@Wafflespeanut Fixed :)

**Comment by jdm** [comment]
I noticed the errors reported by travis seemed to have the expected and actual values backwards. Please verify that the errors make sense before merging this!

**Comment by iamrohit7** [comment]
@jdm If that's from yesterday, I fixed those errors and squashed the commit.

**Comment by jdm** [comment]
@iamrohit7 I know the errors were fixed, but do the messages for new instances of the errors make sense?

File: python/tidy/servo_tidy/tidy.py [hunk 2]
```diff
@@ -510,6 +513,28 @@ def check_rust(file_name, lines):
...
+                sorted_features = sorted(features)
+                if sorted_features != features:
+                    yield(idx + 1, decl_message.format("feature attribute")
+                          + decl_expected.format(tuple(sorted_features))
+                          + decl_found.format(tuple(features)))
+
+                if prev_feature_name > sorted_features[0]:
+                    yield(idx + 1, decl_message.format("feature attribute")
+                          + decl_expected.format(sorted_features[0] + " after " + prev_feature_name)
+                          + decl_found.format(prev_feature_name + " before " + sorted_features[0]))
```
Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: Sorry I didn't notice this before (@jdm has a keen eye!). Linguistic error: Both the lines mean the same thing :smile: 

`Expected bar after foo`
`Found foo before bar`

Comment (iamrohit7) [inline python/tidy/servo_tidy/tidy.py]: I thought that's what the expected output and the actually found ones are. If you have any better ideas, I'll be happy to change.

Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: If we've found `foo before bar`, then how can we expect `bar after foo`? It's already after `foo`, right? Just swap `sorted_features[0]` and `prev_feature_name` in the first line, and we'll be good to go :)

Comment (iamrohit7) [inline python/tidy/servo_tidy/tidy.py]: Done :)

**Comment by Wafflespeanut** [comment]
Thanks!

@bors-servo r+

**Pull request merged**